siolib/scaleio.py

```diff
@@ -844,3 +876,69 @@ def list_volume_infos(self, filters=None):
                               'name': volume['name']})
 
         return infos
+
+    def list_protection_domain_infos(self):
+        """
+        Returns list of protection domains in the system.
+        Each item in the list will include the name and id of the protection domain
+        :return: list of protection domain dictionaries, containing 'name' and 'id'
+        """
+
+        infos = []
+        r_uri = '/api/types/ProtectionDomain/instances'
+        req = self._get(r_uri)
+        if req.status_code != 200:
+            raise exceptions.Error('Error listing protection domains: %s'
+                                   % req.json().get('message'))
+
+        pd_objects = req.json()
+        for pd in pd_objects:
+            infos.append({'id': pd['id'],
+                          'name': pd['name']})
+
+        return infos
+
+    def list_storage_pool_infos(self, protection_domain):
+        """
+        Returns list of storage pools within a protection domain.
+        Each item in the list will include the name and id of the storage pool
+        :param: protection_domain. Name of the protection domain to query for pools
+        :return: list of storage pool dictionaries, containing 'name' and 'id'
+        """
+
+        if not protection_domain:
+            raise ValueError(
+                'Invalid protection_domain parameter, protection_domain=%s'
+                % protection_domain)
+
+        pd_id = self._get_pdid(protection_domain)
+        infos = []
+        r_uri = '/api/instances/ProtectionDomain::' + pd_id + '/relationships/StoragePool'
+        req = self._get(r_uri)
+        if req.status_code != 200:
+            raise exceptions.Error('Error listing storage pools: %s'
+                                   % req.json().get('message'))
+
+        sp_objects = req.json()
+        for sp in sp_objects:
+            infos.append({'id': sp['id'],
+                          'name': sp['name']})
+
+        return infos
+
+    def get_scaleio_api_version(self):
+        """
+        Returns ScaleIO REST API version.
+        This may or may not match the version of ScaleIO.
+        :return: String containing version number
```

Comment:
Probably the description should be more specific. "2.0", '2.0', v2.0.1 - all these strings contain version number. But as a user of the API i want to have more detailed specification.